.github/workflows/release.yml

```diff
@@ -12,22 +12,90 @@ jobs:
-          git tag -a v${{ steps.release.outputs.major }}.${{ steps.release.outputs.minor }} -m "Release v${{ steps.release.outputs.major }}.${{ steps.release.outputs.minor }}"
-          git push origin v${{ steps.release.outputs.major }}
-          git push origin v${{ steps.release.outputs.major }}.${{ steps.release.outputs.minor }}
+        if: steps.release.outputs.releases_created == 'true'
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        env:
+          RELEASES: ${{ toJSON(steps.release.outputs) }}
+        with:
+          script: |
+            const createOrUpdateTag = async (tag, sha) => {
+              const ref = `refs/tags/${tag}`;
+              let existingRef = null;
+              try {
+                const { data } = await github.rest.git.getRef({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  ref
+                });
+
+                existingRef = data;
+              } catch (e) {
+                if (e.status !== 404) {
+                  throw e;
+                }
+              }
+
+              if (existingRef) {
+                console.log(`Updating tag ${tag} from ${existingRef.object.sha} to ${sha}`);
+                await github.rest.git.updateRef({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  ref,
+                  sha
+                });
+              } else {
+                console.log(`Creating tag ${tag} at ${sha}`);
+                await github.rest.git.createRef({
+                  owner: context.repo.owner,
+                  repo: context.repo.repo,
+                  ref,
+                  sha
+                });
+              }
+            }
+
+            const rsplit = (str, sep, maxsplit) => {
+              var split = str.split(sep);
+              return maxsplit ?
+                [split.slice(0, -maxsplit).join(sep)].concat(split.slice(-maxsplit)) :
+                split;
+            }
+
+            const releases = JSON.parse(process.env.RELEASES);
+
+            // Filter `releases` to get the `*--release_created` outputs where
+            // the value is `"true"`. Then strip off that suffix to get an array
+            // of components that were released.
+            const components = Object.entries(releases)
+              .filter(([key, value]) => key.endsWith('--release_created') && value === 'true')
+              .map(([key]) => key.replace(/--release_created$/, ''));
+
+            console.log(`Components released: ${components.join(', ')}`);
+
+            for (const component of components) {
+              const tag = releases[`${component}--tag_name`];
+              const sha = releases[`${component}--sha`];
+
+              console.log(`Updating major and minor tags for ${component} to ${sha}`);
+
+              const [name, semver] = rsplit(tag, '-', 1);
```

Comment:
**question**: would the separator need to be `--`?

Not exactly sure what the `rsplit` would give return here, but based on the python definition of `rsplit`, wouldn't the `name` return `${component}-`?

Reply:
ah nah, this is a different thing we're splitting on.

`--` is a separator in the names of the outputs which the action creates. _keys_ of the outputs object.

This is something else - the _tag_ the action pushed which is a value associated to one of those keys. That's something like `aws-auth-v1.0.0`. So this `rsplit` would give you `["aws-auth", "v1.0.0"]`. Check logs from a test run [here](https://github.com/grafana/shared-workflows/actions/runs/11049876522/job/30696401275). 👍

Reply:
Ahhhhhhh right. Thanks!